apps/Forge/install

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+version=forge-1.6.64
+subversion=${version/forge-} # remove "forge-" from string
+
+game_folder="$HOME/.local/share/mtg-forge"
+repo_url="https://github.com/card-forge/forge/releases/download/${version}"
+archive_name="forge-gui-desktop-${subversion}.tar.bz2"
+
+#cleanup
+trap "rm -f /tmp/$archive_name; exit" EXIT
+
+status "Installing Java 17"
+case "$__os_id" in
+# Raspbian is not reported as a derivative of Debian (no /etc/upstream-release/lsb-release file)
+Raspbian | Debian | Kali)
+  adoptium_installer || exit 1
+  install_packages temurin-17-jre || exit 1
+  ;;
+Ubuntu)
+  install_packages openjdk-17-jre || exit 1
+  ;;
+*)
+  error "$__os_id appears to be an unsupported OS"
+  ;;
+esac
+
+status "Downloading snapshot archive."
+[[ -f "/tmp/$archive_name" ]] && rm -f "/tmp/$archive_name"
+wget "$repo_url/$archive_name" --directory-prefix /tmp || error 'Failed to download latest release!'
+
+# check if folder already exists
+if [[ -d "$game_folder" ]]; then
+  warning "Found existing Forge directory in $game_folder !!!" && echo "It will be moved to /tmp/mtg-forge-old !!!"
+  mv 	"$game_folder" /tmp/MTG-Forge-old || error "Failed to move $game_folder directory to /tmp/ !!! Please, move or delete it yourself and then try installing again."
+fi
+
+mkdir	"$game_folder"
+
+status "Extracting the game archive."
+tar -xvf "/tmp/forge-gui-desktop-${subversion}.tar.bz2" --directory="$game_folder" || error 'Failed to extract the game archive!'
+
+# problem: forge-adventure doesn't launch with java versions > 8
+# workaround: create `forge-mobile.sh` script that's a modified copy of `forge.sh` script
+status "Creating workaround for adventure shell script."
+jarname="$(find "$game_folder" -name 'forge-gui-mobile*.jar' | xargs basename)"
+cp "$game_folder/forge.sh" "$game_folder/forge-mobile.sh"
+sed -i "s|forge-gui-desktop-.*-jar-with-dependencies.jar|$jarname|" "$game_folder/forge-mobile.sh"
```

Comment:
What is the goal here? if the program doesn't launch with java > 8 then why are you installing java 17?

Reply:
> with java > 8 then why are you installing java 17?

the app itself does work with java version > 8, but iirc launch scripts for "adventure" mode doesn't have proper flags so it just fails.

Why not just install java 8? Because it doesn't work with java 8 😂. Forge needs a java fx library that's not available in opensource builds of java 8, apparently it has support only for official oracle's build of java 8 and that's certainly harder to legally obtain. 

Maybe it's possible to make it work with java 8 on linux, but with my limited knowledge of java - I couldn't. But all other versions of java should be supported, I chose java 17 because most software in pi-apps use that version, so good chances that it will be already installed.

Reply:
Can you link an upstream bug report for that issue (or PR to fix it if one is open). If it's just bad flags it would be better to have it corrected upstream.

Reply:
> Can you link an upstream bug report for that issue (or PR to fix it if one is open). If it's just bad flags it would be better to have it corrected upstream.

Here's my issue from 1+ year ago - https://github.com/Card-Forge/forge/issues/2221, it was marked as stale and closed. Note, that I was absolutely unfamiliar with git at that time.  
Looking at it now, I could try to submit PR that fixes the issue and we will see if it gets somewhere 😅.